linkerd/http-retry/src/lib.rs

```diff
@@ -301,31 +311,29 @@ where
             && is_inner_eos
     }
 
-    fn size_hint(&self) -> http_body::SizeHint {
-        let mut hint = http_body::SizeHint::default();
-        if let Some(ref state) = self.state {
-            let rem = state.buf.remaining() as u64;
-
-            // Have we read the entire body? If so, the size is exactly the size
-            // of the buffer.
-            if state.is_completed {
-                return http_body::SizeHint::with_exact(rem);
-            }
-
-            // Otherwise, the size is the size of the current buffer plus the
-            // size hint returned by the inner body.
-            let (rest_lower, rest_upper) = state
-                .rest
-                .as_ref()
-                .map(|rest| {
-                    let hint = rest.size_hint();
-                    (hint.lower(), hint.upper().unwrap_or(0))
```

Comment:
I think that `hint.upper().unwrap_or(0)` is a subtle bug -- if the lower bound is non-zero, we'll end up having an upper limit that is less than the lower limit, which is bad.